src/docs/arch-metadata.dox

```diff
@@ -1,19 +1,78 @@
-configuration string.  The configuration itself is a list of key/value
-pairs in string form.  The configuration's keys are dependent on the
-type of \c uri.  Thus, a metadata entry with a \c uri key beginning with
-\c "table:" will be a configuration string
-having configuration entries like \c key_format and \c value_format to
-describe the data encoding for the uri.  A metadata key beginning with
+Metadata in WiredTiger captures important information about the users database.
+Metadata includes tracking essential information such as the files and tables
+present in the database, their associated configurations, as well as the latest checkpoints of
+each those files.
+
+The main metadata in WiredTiger is stored in the \c WiredTiger.wt file. The
+\c WiredTiger.wt file being a table itself.
+
+@section metadata_table Metadata Table
+
+The keys in the \c WiredTiger.wt table are \c uri strings. The \c uri keys can be strings
+representing:
+- WiredTiger Tables and Files: \c "table:" & \c "file:"
+- Indexes: \c "index:"
+- Colgroups: \c "colgroup:"
+- System values: \c "system". Representing special system values.
+
+The value corresponding to a given \c uri key in the metadata table is its
+associated configuration string.  The configuration value itself is a list of key/value
+pairs in string form.  The key/value pairs in the configuration strings are dependent on the
+type of \c uri.  Thus, a metadata entry with a \c uri key beginning with \c "table:" will be
+a configuration string having entries such as \c key_format and \c value_format to
+describe the data encoding for the uri. A metadata key beginning with
 \c "file:" will have a different set of configuration entries associated
 with it.
 
-A caller of WiredTiger can use WT_SESSION::open_cursor with a \c uri equal to
-\c "meta:" to read the metadata.  Using this interface, metadata can only
-be queried, not changed.  Changes to the metadata are affected by API calls
-such as WT_SESSION::create, WT_SESSION::drop, WT_SESSION::rename.
+@section metadata_checkpoint Storing Checkpoint Metadata
+
+Metadata plays an important role in the WiredTiger checkpointing process. Metadata is used to store
+the latest checkpoint information for each btree in the system. The checkpoint metadata providing a known
+point in time in which WiredTiger can recover from (e.g after a crash or unexpected shutdown).
+The checkpoint metadata for a given btree is stored within the configuration string under its
+associated \c uri key within the \c WiredTiger.wt table. Metadata configuration entries relevant to capturing
+a given btrees checkpoint state include:
+- \c checkpoint: The files checkpoint entries, including their timestamp, transaction ID and write generation.
+- \c checkpoint_lsn: The LSN of the last checkpoint.
+- \c checkpoint_backup_info:  Incremental checkpoint backup information.
+
+Additionally, the metadata table \c WiredTiger.wt is a btree itself that needs to be checkpointed. Since a
```

Comment:
I suggest moving this paragraph to be the first paragraph of the turtle section but remove `Additionally,`. Then you can remove overlapping information from the other paragraph below.

Reply:
That makes sense. I've reworked this paragraph into the `Turtle File` section: bea340fa67e371c75430cfb03b6111aa89b09b46